pkg/kubectl/bind-apiservice/plugin/servicebindings_test.go

```diff
@@ -0,0 +1,756 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	kubebindv1alpha1 "github.com/kube-bind/kube-bind/pkg/apis/kubebind/v1alpha1"
+)
+
+func TestHumanReadablePromt(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name           string
+		testData       kubebindv1alpha1.PermissionClaim
+		expectedOutput string
+	}{
+		{"Owner=Provider",
+			kubebindv1alpha1.PermissionClaim{
+				GroupResource: kubebindv1alpha1.GroupResource{
+					Group:    "",
+					Resource: "foo",
+				},
+				Version: "v1",
+				Selector: kubebindv1alpha1.ResourceSelector{
+					Owner: kubebindv1alpha1.Provider,
+				},
+				Required: true,
+			},
+			"Create and modify provider owned foo objects (apiVersion: \"v1\") on your cluster.\n" +
+				"Accepting this Permission is required in order to proceed.\n" +
+				"Do you accept this Permission? [No,Yes]\n",
+		},
+		{"Owner=Provider,Required=false",
+			kubebindv1alpha1.PermissionClaim{
+				GroupResource: kubebindv1alpha1.GroupResource{
+					Group:    "",
+					Resource: "foo",
+				},
+				Version: "v1",
+				Selector: kubebindv1alpha1.ResourceSelector{
+					Owner: kubebindv1alpha1.Provider,
+				},
+				Required: false,
+			},
+			"Create and modify provider owned foo objects (apiVersion: \"v1\") on your cluster.\n" +
+				"Accepting this Permission is optional.\n" +
+				"Do you accept this Permission? [No,Yes]\n",
+		},
+		{"Owner=Provider,Selector.Name",
+			kubebindv1alpha1.PermissionClaim{
+				GroupResource: kubebindv1alpha1.GroupResource{
+					Group:    "",
+					Resource: "foo",
+				},
+				Version: "v1",
+				Selector: kubebindv1alpha1.ResourceSelector{
+					Name:  "bar",
+					Owner: kubebindv1alpha1.Provider,
+				},
+				Required: true,
+			},
+			"Create and modify the following provider owned foo objects (apiVersion: \"v1\") on your cluster referenced with:\n" +
+				"	name: \"bar\"\n" +
+				"Accepting this Permission is required in order to proceed.\n" +
+				"Do you accept this Permission? [No,Yes]\n",
+		},
+		{"Owner=Provider,GroupResource.Group",
+			kubebindv1alpha1.PermissionClaim{
+				GroupResource: kubebindv1alpha1.GroupResource{
+					Group:    "bar",
+					Resource: "foo",
+				},
+				Version: "v1",
+				Selector: kubebindv1alpha1.ResourceSelector{
+					Owner: kubebindv1alpha1.Provider,
+				},
+				Required: true,
+			},
+			"Create and modify provider owned foo objects (apiVersion: \"bar/v1\") on your cluster.\n" +
```

Comment:
**note**: the object is provider owned because create.donate is not specified. Not because the selector is owner=provider. The selector owner=provider just means that the reconciler will only look at provider owned objects (if a consumer owned object exists, the reconciler will skip it)

Reply:
Reworked this in 926bb5993775516f6dbfe28787228a9b11d95f92.